lib/vanagon/component/dsl.rb

```diff
@@ -289,18 +289,44 @@ def environment(env)
         @component.environment.merge!(env)
       end
 
-      # Adds actions to run at the beginning of a package install
+      # Adds action to run during the preinstall phase of packaging
       #
-      # @param action [String] Bourne shell compatible scriptlets to execute
-      def add_preinstall_action(action)
-        @component.preinstall_actions << action
+      # @param pkg_state [Array] the state in which the scripts should execute. Can be
+      #   one or multiple of 'install' and 'upgrade'.
+      # @param block [Array] the Bourne shell compatible scriptlet(s) to execute
+      def add_preinstall_action(pkg_state, block)
```

Comment:
Is there a case where we would want to remove a packaging action? Add to me makes the most sense (i.e., https://github.com/melissa/razor-vanagon/blob/master/configs/components/razor-server.rb#L86-L90). It follows established convention, like `install_configfile`, since it can be called multiple times